src/main/scala/org/apache/mesos/chronos/scheduler/jobs/Schedule.scala

```diff
@@ -0,0 +1,44 @@
+package org.apache.mesos.chronos.scheduler.jobs
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import org.joda.time.{DateTime, Period}
+
+
+case class Schedule(@JsonProperty schedule: String,
+                    @JsonProperty scheduleTimeZone: String,
+                    @JsonProperty invocationTime: DateTime,
+                    @JsonProperty originTime: DateTime,
+                    @JsonProperty offset: Long = 0,
+                    @JsonProperty recurrences: Option[Long] = None,
+                    @JsonProperty period: Period) {
+
+
+  def next: Option[Schedule] = {
+    if(recurrences.exists(_ == 0)) None
+    else {
+      val nextOffset = offset + 1
+      val nextInvocationTime = Schedule.addPeriods(originTime, period, nextOffset.asInstanceOf[Int] /* jodatime doesn't support longs in their plus methods. What to do about overflow? */)
```

Comment:
That's a good point, although 2147483647 seconds is about 68 years. I think we should log such overflow cases and cap them at Int.MaxValue for now.